feat(async-config) 스레드 풀 조정

yokxim2 opened the pull request:
## ✨ 담당 파트
- 비동기로 호출될 스레드 풀 조정

## 🔎 작업 상세 내용
- 주석을 참고해주세요.
- https://devoong2.tistory.com/entry/Spring-Async-%EC%82%AC%EC%9A%A9-%EB%B0%A9%EB%B2%95-%EB%B0%8F-TaskExecutor-ThreadPool
이 링크를 참고했습니다.

## 🔧 앞으로의 과제
- 이벤트/메시지 브로커 적용 예정

## ✅ 테스트 코드 작성 및 기능 테스트 여부
- [] 테스트 코드 작성
- [] 기능 테스트 여부

## ➕ 이슈 링크
- #44

yokxim2 committed:
feat(async-config) 스레드 풀 조정 #44

- 주석을 참고해주세요.

hubaek reviewed (approved):
뚝딱뚝딱 금방하는 코드 LGTM

for2ver0 commented on src/main/java/com/summoner/lolhaeduo/client/event/AsyncConfig.java:
```diff
@@ -1,9 +1,26 @@
+    public Executor getAsyncExecutor() {
+        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+        executor.setCorePoolSize(5);                // Thread pool에서 기본적으로 유지되는 스레드의 개수
+        executor.setMaxPoolSize(10);                // Thread pool에서 사용할 수 있는 최대 스레드의 개수
```
- MaxPoolSize의 값을 10으로 설정한 것은 어떠한 기준으로 설정한 건지 궁금합니다.
- 임의로 값을 설정하신 걸까요? 최적 값을 계산해서 설정하신 걸까요?

ThreadPool 값 설정에 대해서는 아래 블로그를 참고하시면 좋을 것 같습니다.
[https://velog.io/@vanillacake369/Async-Size-%EC%84%A4%EC%A0%95-%EA%B8%B0%EC%A4%80%EC%97%90-%EB%8C%80%ED%95%B4-%EA%B3%A0%EB%AF%BC%ED%95%B4%EB%B3%B4%EC%9E%90-feat.ThreadPoolQueue](url)

for2ver0 commented on src/main/java/com/summoner/lolhaeduo/client/event/AsyncConfig.java:
```diff
@@ -1,9 +1,26 @@
+        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+        executor.setCorePoolSize(5);                // Thread pool에서 기본적으로 유지되는 스레드의 개수
+        executor.setMaxPoolSize(10);                // Thread pool에서 사용할 수 있는 최대 스레드의 개수
+        executor.setQueueCapacity(100);             // 작업 큐가 최대로 가질 수 있는 작업 개수
```
위의 MaxPoolSize와 동일한 질문입니다.
- QueueCapacity의 값을 100으로 설정한 것은 어떠한 기준으로 설정한 건가요?

for2ver0 reviewed (commented):
리뷰 확인 부탁드립니다

yokxim2 committed:
fix(async-config) config 설정값 변경 #44

t2 micro 서버를 사용하는 상황에서 CPU 코어 수는 1개이므로 나온 계산입니다.
corePoolSize = CPU 코어 수 * 2
maxPoolSize = corePoolSize * 2 ~ 3
queueCapacity = 비동기로 호출되는 작업 수

queueCapacity의 경우 일반적인 duo.gg 사이트에서 이용하는 사람들의 평균 수인 150명을 커버할 수 있는 숫자로 대체했습니다.

yokxim2 committed:
fix(async-config) config 설정값 변경 #44

queueCapacity 설정값을 기본값으로 유지하도록 변경했습니다.

for2ver0 committed:
fix(async-config) thread-pool 설정값 변경 #44

- CorePoolSize 측정 공식에 따른 안정적인 CorePoolSize 값 설정
- MaxPoolSize 값 설정 삭제
  - QueueCapacity 값 설정 삭제에 따른 MaxPoolSize 값 설정 의미가 없어짐

for2ver0 reviewed (approved):
마지막에 제가 수정했는데 제가 approve 할 수 있는 건가유..?
이게 되네..?
